tests/integration/test_cncf_conformance.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests."""
+
+import logging
+import re
+
+import pytest
+from helpers import get_kubeconfig, ready_nodes, run_command, sonobuoy_tar_gz
+from juju import model, unit
+from pytest_operator.plugin import OpsTest
+
+log = logging.getLogger(__name__)
+
+
+# This pytest mark configures the test environment to use the Canonical Kubernetes
+# bundle for cncf conformance testing, for all the test within this module.
+pytestmark = [
+    pytest.mark.bundle(file="test-bundle.yaml", apps_local=["k8s", "k8s-worker"]),
+    pytest.mark.run_with_k,
```

Comment:
This feels like a hack to me. Have you considered using tags as we do for the k8s-snap? Adding a "long-running" or "conformance" tag or something like this to the test seems more transparent. Right now, I can only guess why we mark this with `run_with_k` (long-running task).

Reply:
@rapour take a look at `test_openstack` and how `pytest.mark.cloud("openstack")` works to skip the tests on certain clouds. 


the same strategy should be here. 


the collection of tests will pick up there is a `test_cncf_conformance` but we can mark the tests as "skipped" because maybe the `--sonobuoy-version` was unset?  


I'm envisioning a change here to be:

```suggestion
    pytest.mark.cncf_conformance,
```

Then create an autouse fixture in conftest.py to require a certain flag to be present otherwise it skips the test.